src/modules/esl-utils/misc/array.ts

```diff
@@ -29,13 +29,44 @@ export const uniq = <T> (arr: T[]): T[] => {
   return result;
 };
 
-/** Crete an array filled with the range [0,..,N-1] */
+/** Create an array filled with the range [0,..,N-1] */
 export function range(n: number): number[];
-/** Crete an array filled with values returned by the filler callback */
+/** Create an array filled with values returned by the filler callback */
 export function range<T>(n: number, filler: (i: number) => T): T[];
 export function range(n: number, filler: (i: number) => any = identity): any[] {
   const arr = Array(n);
   let i = 0;
   while (i < n) arr[i] = filler(i++);
   return arr;
 }
+
+/** Create an array of unique values that presented in each of the passed arrays */
+export function intersection<T>(...rest: T[][]): T[];
+export function intersection(a: any[], b: any[], ...rest: any[][]): any[] {
+  if (rest.length) return intersection(a, intersection(b, ...rest));
+  return a.filter(Set.prototype.has, new Set(b));
+}
+
+/** Create an array with the unique values from each of the passed arays */
+export function union(...rest: any[][]): any[] {
+  const set = new Set();
+  rest.forEach(item => item.forEach(i => set.add(i)));
+  const result: any[] = [];
+  set.forEach(value => result.push(value));
+  return result;
+}
+
+/** Creates an array of unique values from the first array that are not present in the other arrays */
+export function complement<T>(...rest: T[][]): T[];
+export function complement(a: any[], b: any[], ...rest: any[][]): any[] {
```

Comment:
Cosmetic: it will be better to make original sydnature strict (with generic type) until it's really necessary to disable type check with `any`

Reply:
She same for othe r methods